src/inmanta/server/services/orchestrationservice.py

```diff
@@ -885,7 +895,9 @@ async def _put_version(
             for agent in all_agents:
                 await self.agentmanager_service.ensure_agent_registered(env, agent, connection=connection)
 
-            await self._register_agent_code(partial_base_version, version, env.id, module_version_info, connection)
+            await self._register_agent_code(
+                partial_base_version, version, env.id, module_version_info, connection, bypass_base_version_check
```

Comment:
We have a sort of unwritten convention that `connection` is always the last parameter. That said, I don't think we should ever rely on it, and imo it should always be kw-only. I care less about the other options, but I think the bypass option is also a good candidate for kw-only.